test/core/end2end/tests/connectivity.cc

```diff
@@ -33,6 +33,12 @@ typedef struct {
   grpc_completion_queue* cq;
 } child_events;
 
+typedef struct {
+  grpc_experimental_completion_queue_functor functor;
+  bool done;
+  gpr_mu mu;
```

Comment:
If the only purpose of this mutex is to protect `done`, then I suggest just making `done` an atomic.  You can use `grpc_core::Atomic<bool>` for that.

https://github.com/grpc/grpc/blob/master/src/core/lib/gprpp/atomic.h

But see my comment below as to whether synchronization is actually needed here.

Reply:
thanks for the advice I will use the atomic type if I have to protect just one variable!